### DISCUSSION TITLE
How Can I Make A TShock Server?

### DISCUSSION THREAD

**Discussion by Lizena2314** [Pryaxis/TShock#2903] [Q&A] [answered]
Hello, there may be typos in this article.

I can open a tshock server normally but I want to edit the codes and translate them into my language so I downloaded this file but I have no idea what to do.

When I try to open the "TshockLauncher.csproj" and "TshockInstaller.csproj" files, the Visual Studio Code program always opens and I can't do anything at all.

Could you please help me?
![image](https://user-images.githubusercontent.com/75311306/213858385-505cf00d-8645-47d5-8292-7ac82e6e1049.png)

**Comment by sgkoishi** [comment]
To translate, https://crowdin.com/project/tshock
To run it localized, add `-lang <id>` command line args.
List of available Terraria languages:
```
1 English
2 German
3 Italian
4 French
5 Spanish
6 Russian
7 Chinese
8 Portuguese
9 Polish
```

**Reply by Lizena2314** [reply]
And i want to edit codes.

**Reply by sgkoishi** [reply]
Is there something wrong? Visual Studio Code _is_ one of the commonly used editors.

**Reply by Lizena2314** [reply]
Nope, i just asked how i open a server with these files.

**Reply by sgkoishi** [reply]
> If you want to run the `TShockLauncher` (which runs a server), run:
> 1. `dotnet run --project TShockLauncher`

https://github.com/Pryaxis/TShock#developing-tshock

**Reply by Lizena2314** [reply]
Where should i write this code?

**Reply by sgkoishi** [reply]
This is a command to run it, not code to write somewhere.
If you don't know what code and where to write, https://learn.microsoft.com/en-ca/dotnet/fundamentals/

**Reply by Lizena2314** [reply]
![image](https://user-images.githubusercontent.com/75311306/213859724-da18da8f-34cd-4539-92cf-9ff374543e7a.png)

**Reply by Lizena2314** [reply]
No project file.

**Reply by sgkoishi** [reply]
Because you `cd TShockLauncher` for this step while the instruction didn't say so.
It might be better to [search](https://www.google.com/search?q=MSBUILD+%3A+error+MSB1009) first.

**Reply by Lizena2314** [reply]
I tryed to write "dotnet build" and i see these crashes.
![image](https://user-images.githubusercontent.com/75311306/213860317-9e88a123-ac36-44c1-8164-f39ced2c2831.png)

**Reply by sgkoishi** [reply]
Read the instruction.
> 1. Clone the repository: `git clone https://github.com/Pryaxis/TShock.git --recurse-submodules`

**Reply by Lizena2314** [reply]
![image](https://user-images.githubusercontent.com/75311306/213860511-d920ad17-e27a-41de-8278-3536003d2d7e.png)

**Reply by sgkoishi** [reply]
This error message is indicating that the git command is not recognized as a valid command on your computer. This may be because git is not installed on your computer or it may not be added to your system's PATH environment variable. To fix this, you can download and install git from the official website https://git-scm.com/downloads and then add it to your system's PATH.
Alternatively, you can use Git Bash which is a command line tool that comes with git installation on windows and you can use git commands in Git Bash.

<sup>This answer is generated by [ChatGPT](https://openai.com/blog/chatgpt/) by OpenAI.</sup>

**Reply by limitless4321** [reply]
HOW DO I EVEN LOAD IT UP????
(TSHOCK)

**Comment by Lizena2314** [comment, marked as answer]
I did everything but i get the same error again
![image](https://user-images.githubusercontent.com/75311306/213861933-6ba53ea4-cba5-4c7c-b0dc-88fc5543e28a.png)

**Reply by sgkoishi** [reply]
https://github.com/Pryaxis/TShock/discussions/2903#discussioncomment-4743997

**Reply by Lizena2314** [reply]
I've been doing research for about 1 hour, but no source has any error solutions specified and I'm not very understanding of these things, can you help?

**Reply by sgkoishi** [reply]
As the link said, you should run `dotnet run --project TShockLauncher` OUTSIDE `TShockLauncher` folder. If you want to run it inside `TShockLauncher` folder, run `dotnet run`.

**Reply by Lizena2314** [reply]
Oh its worked. Thank you so much for your quick response and help with my questions.
